src/smt/smt_mode.h

```diff
@@ -0,0 +1,57 @@
+/*********************                                                        */
+/*! \file smt_mode.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Andrew Reynolds
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2020 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Enumeration type for the mode of an SmtEngine.
+ **/
+
+#include "cvc4_public.h"
```

Comment:
Can we make this private already or does this break things?

Reply:
I tried :) 

Unfortunately it breaks things (spent an hour trying to fix it but couldn't).  I think we really should make SmtEngine private sooner than later, which would make it easy to make this private.